packages/compass-indexes/src/modules/create-index/new-index-field.ts

```diff
@@ -23,7 +25,7 @@ export const INITIAL_STATE = null;
  *
  * @returns {String} The new state.
  */
-export default function reducer(state = INITIAL_STATE, action) {
+export default function reducer(state = INITIAL_STATE, action: AnyAction) {
```

Comment:
Written like this, 'CREATE_NEW_INDEX_FIELD' and 'CLEAR_NEW_INDEX_FIELD' would refer to a value but be being used as a type here. It would work with `type: string`, but in this case, there would be not so much advantage compared with AnyAction usage.